src/MongoDB.Driver/IndexKeysDefinitionBuilder.cs

```diff
@@ -454,11 +493,13 @@ internal sealed class DirectionalIndexKeyDefinition<TDocument> : IndexKeysDefini
     {
         private readonly FieldDefinition<TDocument> _field;
         private readonly SortDirection _direction;
+        private readonly bool _usingWildcard;
 
-        public DirectionalIndexKeyDefinition(FieldDefinition<TDocument> field, SortDirection direction)
+        public DirectionalIndexKeyDefinition(FieldDefinition<TDocument> field, SortDirection direction, bool usingWildcard = false)
```

Comment:
I don't think this class should be changed at all, because fields in a wild card index don't have an ascending/descending direction.

You could add a new class if necessary.

Reply:
yep, my mistake. Moved a new logic to `WildcardIndexKeyDefinition`